### DISCUSSION TITLE
Config to disable auto-refresh of the dag tree and graph views?

### DISCUSSION THREAD

**Discussion by johnson-jay-l** [apache/airflow#22538] [Q&A]
I'm looking for a way to pass a config to the dag object that defaults the auto-refresh toggle on the UI to the 'off' position.  Or similarly, to configure the `auto_refresh_interval` at the dag level.

The auto-refresh feature on the tree and graph views is nice but for some of our larger dags, it can't fetch the updates within the refresh interval and the UI becomes almost unresponsive.  The dag in question has just under 2,000 tasks (I won't go into details about why there are so many tasks, in short they are code-generated tasks).

An option is to increase the value of [auto_refresh_interval](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#auto-refresh-interval), but that will cause users to wait longer for refreshes in smaller dags where it isn't an issue.  And of course making the dag smaller is something we've considered but that comes with its own set of challenges and tradeoffs.

The code for [graph.html](https://github.com/apache/airflow/blob/2b4bf7fe67fc656ceb7bdaad36453b7a5b83ef04/airflow/www/templates/airflow/graph.html#L114) defaults the toggle to 'on' when the dag is running.

Could an arg be passed into the dag's default_args, then parsed in the html?  Something like:

```
default_args = {
    'owner': 'dataeng',
    'depends_on_past': False,
    'start_date': datetime.strptime(START_DATE, '%Y-%m-%d %H:%M:%S'),
    'on_failure_callback': failure_callback,
    'retries': 2,
    'retry_delay': timedelta(hours=1),
    'max_active_runs': 1,
    'params': {
        'auto_refresh_disabled': True
    }
}
```

Then in the graph.html, add something like `... and not dag.auto_refresh_disabled` to the checkbox state:
```
<input class="switch-input" id="auto_refresh" type="checkbox" {% if dag_run_state == 'running' and not dag.params.auto_refresh_disabled %}checked{% endif %}>
```

Thanks!

edit: fixed the conditional check logic in the mocked examples

edit2: looks like params is the way to pass custom args into the default_args

**Comment by walterdolce** [comment]
I also found myself wanting to disable auto-refresh. A DAG with hundreds if not thousands of tasks to run will cause the page to hang forever and the auto-refresh feature exacerbates the problem.

**Comment by potiuk** [comment]
Create a PR and discuss it there. Looks like nice and simple thing to do and when you discuss it with a code it's easier to iterate on and get merged when you complete it.

**Comment by bbovenzi** [comment]
Right now, disable auto-refresh is kept in localStorage but is a single key across all DAGs. We could at least have a dag_id attached to the key and have a localStorage key per disabled dag. That would still require a user to manually set that on their browser. I also don't think DAG code should be concerned with how the UI works. This should probably work similar to a DAG's `is_paused`. 

Another thought, we should also make auto-refresh work better with large DAGs. I'd like to get to the point that an autorefresh endpoint only ever returns what is new and not the entire DAG every time. The new Grid view tries to do this on the UI-side but every API request is still a large JSON object that needs to be diff-ed.
